src/hilbert_tree.h

```diff
@@ -138,6 +140,33 @@ struct bvh {
     return std::views::iota((uint32_t)first, (uint32_t)last);
   }
 
+  static constexpr T node_width(aabb<T, N> b) {
+    auto l = b.lengths();
+    if constexpr(N == 2) {
+      return std::max(l[0], l[1]);
+    } else {
+      return std::max(std::max(l[0], l[1]), l[2]);
+    }
+  }
+
+  static constexpr T mass(monopole m) {
+    return m[3];
```

Comment:
This is only correct for 3D right? I think it should be `m[N]`?

Reply:
```suggestion
    return m[N];
```

Reply:
Oopsie, good catch, thank you!